src/Core__Type.resi

```diff
@@ -1,75 +1,65 @@
 /***
-Utilities for classifying the type of JavaScript values at runtime.
+Utilities for classifying and safely using JavaScript values at runtime.
 */
 
 /**
-The possible types of JavaScript values.
+The possible types of JavaScript values. See [typeof on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)
 */
-type t = [#undefined | #object | #boolean | #number | #bigint | #string | #symbol | #function]
+type typeof = [
+  | #undefined
+  | #object
+  | #boolean
+  | #number
+  | #bigint
+  | #string
+  | #symbol
+  | #function
+]
 
 /**
-`typeof(someValue)`
+An abstract type-safe representation for values whose contents are a mystery, or at least not 100% guaranteed to be what you expect. A value of type `unknown` could a number, string, function, null, or any other JavaScript type. To safely use an `unknown` value use the functions in this module to test and classify it as its true underlying type.
+*/
+type unknown
 
-Returns the underlying JavaScript type of any runtime value.
+external toUnknown: 'a => unknown = "%identity"
 
-See [`typeof`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof) on MDN.
+/**
+`typeof(value)` returns the underlying JavaScript type of `value` at runtime. **Note** null values return `object`. See [typeof on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)
 
 ## Examples
 ```rescript
-Console.log(Type.typeof("Hello")) // Logs "string" to the console.
-
-let someVariable = true
-
-switch someVariable->Type.typeof {
-| #boolean => Console.log("This is a bool, yay!")
-| _ => Console.log("Oh, not a bool sadly...")
-}
+Type.typeof("Hello") // "string"
+Type.typeof(3) // "number"
+Type.typeof(3.14) // "number"
+Type.typeof(null) // "object"
 ```
 */
-external typeof: 'a => t = "#typeof"
-
-module Classify: {
-  /***
-  Classifies JavaScript runtime values.
-  */
-
-  /**
-  An abstract type representing a JavaScript function.
-
-  See [`function`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function) on MDN.
-  */
-  type function
+external typeof: 'a => typeof = "#typeof"
 
-  /**
-  An abstract type representing a JavaScript object.
+// Why can't this be {..} so it can be used by the Object module?
+type object
```

Comment:
`{..}` is not a concrete type. It hides an implicit type variable, sometimes called the row variable, because it can be seen as being a substitute for all the fields of a concrete type that is not explicitly specified. It is more accurate to view this as a type variable with an object constraint. It can be anything, as long as it's an object. But it has to be something, and that something is tracked by the type variable.

Reply:
It's better to make only that conversion unsafe, than to make this function as a whole unsound.

Reply:
I don't understand "unsound". If there was a way to classify this unknown thing is a Object('a as {...}) what is unsafe about that? I'm trying to think of ways I can use it dangerously but can't. Rescript will prevent me from using accessors on it. The type definition is "some kind of object". It's not a false statement like when I do get in the Object module and get an option<anything I want it to be> out. I'm not saying it is important we have this ReScript equivalent of an OCaml thing. I just don't understand what your saying.

Reply:
I added safe accessors/getters if you want to take a peek.

Reply:
"Soundness" in type system lingo means it's guaranteed to be correct. This isn't, because `{..}` does not mean "a an object type with unknown fields", but rather "an object type which will be inferred to be whatever you want it to be rather than what it actually is.". 

> Rescript will prevent me from using accessors on it.

No it won't.

```rescript
let x: {..} = Obj.magic({"bar": 42})
let y = x["foo"] + 3
```

Because `x` is used as if it had a `foo` field, the type will be inferred as `{"foo": int}`.

Reply:
Ok now I understand. I hadn't wrapped my head around the idea that `{..}` could be used as if it had anything in it, just like `'a` can be used as if were anything.